docs_parser.py

```diff
@@ -0,0 +1,79 @@
+
+import argparse #we use the argparse module for passing command-line arguments on startup.
+import pdb
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("p", metavar="filepath/to/doc.txt", help="relative or absolute filepath to the document you'd like to test parsing on.")
+    args = parser.parse_args()
+    return args
+
+def parse(filepath):
+    tokens = []
+    with open(filepath) as docfile:
+        index = 0
+        lines = docfile.readlines()
+        while index < len(lines):
+            #decide which kind of token we are looking at
+            if lines[index].strip() == '': #if the next line is blank
+                pass
+            elif lines[index][0:4] == '    ' or lines[index][0] == '\t': #if the next line is an indented paragraph
+                index = append_paragraph(lines, index, tokens)
+            elif lines[index].strip() == '==============================': # if the next line is a Heading
+                append_Heading(lines, index, tokens)
+                index = index + 1 # skip the second line of equals sings
```

Comment:
sings -> signs.
Not that singing is bad, because it isn't, but it sadly is not within the scope of the parser.

Reply:
I appreciate your sense of humor. Will do.